src/lib.rs

```diff
@@ -486,9 +483,9 @@ mod tests {
         let bytes_from: [u8; 8] = [1, 2, 3, 4, 5, 6, 7, 8];
         let bytes_to = &mut bytes_to[..];
         let bytes_from = &bytes_from[..];
-        for i in 0..bytes_from.len() {
-            bytes_to[i] = bytes_from.pread(i).unwrap();
-        }
+        bytes_to.iter_mut().enumerate().for_each(|(i, item)| {
+            *item = bytes_from.pread(i).unwrap();
+        });
```

Comment:
i think this is a bit hard to read - iterating over the "target" as mutable.  I would suggest a simpler variant (i didn't test it, might not even compile) -- iterate over source values, pushing them into the output array.


```suggestion
        for (idx, value) in bytes_from.iter().enumerate() {
            bytes_to[i] = value.unwrap();
        });
```